sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -2143,7 +2143,10 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
                     bufferSize: options?.BufferSize ?? Constants.DefaultBufferSize,
                     position: position,
                     conditions: conditions,
-                    progressHandler: options?.ProgressHandler);
+                    progressHandler: options?.ProgressHandler,
+                    blobHttpHeaders: options?.HttpHeaders,
+                    metadata: options?.Metadata,
+                    tags: options?.Tags);
```

Comment:
Should we also do this when creating the empty blob above?

Reply:
fixed